share/readonlyadmin.py

```diff
@@ -0,0 +1,56 @@
+## adopted from http://mike.hostetlerhome.com/blog/2012/11/13/add-a-read-only-role-to-django-admin/
+from django.contrib import admin
+from django.core.exceptions import PermissionDenied
+
+
+class ReadOnlyAdmin(admin.ModelAdmin):
+    """
+    """
+
+    def has_add_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def has_delete_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def get_actions(self, request):
+
+        actions = super(ReadOnlyAdmin, self).get_actions(request)
+
+        if self.__user_is_readonly(request):
+            if 'delete_selected' in actions:
+                del actions['delete_selected']
+
+        return actions
+
+    def change_view(self, request, object_id, extra_context=None):
+
+        if self.__user_is_readonly(request):
```

Comment:
I'm getting the below error when programmatically including all of the fields. It only occurs when viewing the details of a celery task.
```
Internal Server Error: /admin/share/celerytask/1723/change/
Traceback (most recent call last):
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/options.py", line 639, in get_form
    return modelform_factory(self.model, **defaults)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/forms/models.py", line 545, in modelform_factory
    return type(form)(class_name, (form,), form_class_attrs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/forms/models.py", line 257, in __new__
    raise FieldError(message)
django.core.exceptions.FieldError: Unknown field(s) (traceback, timestamp) specified for CeleryTask

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/options.py", line 541, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/sites.py", line 244, in inner
    return view(request, *args, **kwargs)
  File "/Users/laurenbarker/GitHub/COS/SHARE/share/readonlyadmin.py", line 32, in change_view
    return super(ReadOnlyAdmin, self).change_view(request, object_id, extra_context=extra_context)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1440, in change_view
    return self.changeform_view(request, object_id, form_url, extra_context)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1367, in changeform_view
    ModelForm = self.get_form(request, obj)
  File "/Users/laurenbarker/.virtualenvs/share/lib/python3.5/site-packages/django/contrib/admin/options.py", line 642, in get_form
    % (e, self.__class__.__name__))
django.core.exceptions.FieldError: Unknown field(s) (traceback, timestamp) specified for CeleryTask. Check fields/fieldsets/exclude attributes of class CeleryTaskAdmin
```